### PR TITLE
feat: use the new opa policies

### PR DESCRIPTION
**Pull request by hsiliev**
Use the new OPA policies in the code

Solves PZ-2176

### CODE REVIEW

**Review by bas-info-nl — APPROVED** [review]

**Review by edgarvonk — APPROVED** [review]
Nice. Had to tell if you did not make any mistakes of course. ;-) As discussed it would be good to start adding more automated tests. Not sure which ones yet. I would prefer not to test authorisation things in our integration tests I think. For later.

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java [hunk 1]
```diff
@@ -280,52 +279,52 @@ public RESTEnkelvoudigInformatieobject createEnkelvoudigInformatieobjectAndUploa
...
             @Valid @MultipartForm final RESTEnkelvoudigInformatieobject restEnkelvoudigInformatieobject
     ) {
-        final Zaak zaak = zrcClientService.readZaak(zaakUuid);
-        assertPolicy(policyService.readZaakRechten(zaak).wijzigen());
+        var zaak = zrcClientService.readZaak(zaakUuid);
+        assertPolicy(policyService.readZaakRechten(zaak).toevoegenDocument());
 
-        final EnkelvoudigInformatieObjectData enkelvoudigInformatieObjectData = taakObject ?
-                informatieobjectConverter.convertTaakObject(restEnkelvoudigInformatieobject) :
+        var enkelvoudigInformatieObjectData = taakObject ? informatieobjectConverter.convertTaakObject(restEnkelvoudigInformatieobject) :
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:285]: Do we have this in Java 21?

File: src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java [hunk 2]
```diff
@@ -280,52 +279,52 @@ public RESTEnkelvoudigInformatieobject createEnkelvoudigInformatieobjectAndUploa
             @QueryParam("taakObject") final boolean taakObject,
             @Valid @MultipartForm final RESTEnkelvoudigInformatieobject restEnkelvoudigInformatieobject
     ) {
-        final Zaak zaak = zrcClientService.readZaak(zaakUuid);
-        assertPolicy(policyService.readZaakRechten(zaak).wijzigen());
+        var zaak = zrcClientService.readZaak(zaakUuid);
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java:282]: I think this is not supported (yet) in Java 21

**Comment by hsiliev** [comment]
Starting from already merged branch breaks the GitHub merge queue

**Pull request closed without merging**